### ISSUE TITLE
Allow using poetry as a dependency management tool (as lambdas support this in the cdk now)

### ISSUE THREAD

**Issue by mcsheehan** [aws/chalice#1769]
Lambdas support using poetry in the cdk now. 
https://docs.aws.amazon.com/cdk/api/latest/docs/aws-lambda-python-readme.html

It would be really nice if the chalice cdk deployment had a way of allowing this too.

Would this involve an alternative packager? Or is there a cdk only solution?

**Comment by ghilesmeddour** [comment]
Any planned support for this?

**Comment by baotran2207** [comment]
Waiting for this feature !

**Comment by mcsheehan** [comment]
@Insighttful yeah you've got it. The CI pipeline is pretty much poetry export, then chalice package, then cloudformation deploy of it.
`
poetry install --no-dev
poetry export --without-hashes > requirements.txt
poetry run chalice package --stage << parameters.env >> --template-format yaml packaged/
aws cloudformation package --template-file ./packaged/sam.yaml --s3-bucket template-bucket --output-template-file cfn-output_sam.yml
`

**Comment by gwendal-toullec** [comment]
+1 for this feature
